kimchi/src/prover.rs

```diff
@@ -632,12 +633,7 @@ where
             }
 
             if let Some(selector) = index.cs.foreign_field_add_selector_poly.as_ref() {
-                index_evals.extend(
-                    foreign_field_add::gadget::circuit_gates()
-                        .iter()
-                        .enumerate()
-                        .map(|(_, gate_type)| (*gate_type, &selector.eval8)),
-                );
+                index_evals.insert(GateType::ForeignFieldAdd, &selector.eval8);
```

Comment:
Suggest to switch back to the "common interface" code, so one day this prover code can be made generic.

Reply:
As above, changed it after Matthew asked me to do it for single gate gadgets

Reply:
Let's go with what Matthew says.